### ISSUE TITLE
Support U24

### ISSUE THREAD

**Issue by mfeit-internet2** [perfsonar/pscheduler#1441]
Title says it all.

TO-DO:

- [x] Do a build with OWAMP present and make sure that all works
- [x] Consolidate macros that determine whether or not to build iperf3.
- [x] Build of `pscheduler-tool-s3-benchmark` doesn't work right because of missing dependency
- [x] Convert all use of Python `re` to use raw strings (`r''`) to dodge invalid escape sequence warnings
- [x] Upgrade jq to 1.7 (See #1445)

**Comment by laeti-tia** [comment]
I just tested a build from the `issue-1441` branch and see a possible problem with `jq`. Ubuntu 24 is shipping `jq-1.7.1` which gets installed instead of our own 1.6.0 build.  And I then get this error:

```
ERROR:  ImportError: /usr/lib/python3/dist-packages/_pyjq.cpython-312-x86_64-linux-gnu.so: undefined symbol: jv_is_integer_large
```

Should we force installing our 1.6.0 or should we build a newer 1.7.1 containing this missing function?

**Comment by mfeit-internet2** [comment]
Neither.  The missing function was added as part of a patch to resolve #717 and #785.  This will be covered by a jq upgrade in #1445.

**Comment by mfeit-internet2** [comment]
@laeti-tia If we specify 1.6 as a dependency, will that be installed even if 1.7 is available?  I'm thinking we should just stick to 1.6 for the bugfix and save the upgrade for the next feature release.

**Comment by laeti-tia** [comment]
> @laeti-tia If we specify 1.6 as a dependency, will that be installed even if 1.7 is available? I'm thinking we should just stick to 1.6 for the bugfix and save the upgrade for the next feature release.

We'll need to add an APT configuration file to make it work, because the version number is lower.  That can be done with the following content in `/etc/apt/preferences.d/99-perfsonar-jq`:

```
Package: jq libjq1
Pin: release perfsonar-5.1*
Pin-Priority: 510
```

**Comment by mfeit-internet2** [comment]
With the completion of #1445, 5.2 will have jq 1.7, so the patch problem goes away and U24 can use the version provided by the distro.

There is now a functioning build on U24, so I'm calling this good.
